### PR TITLE
always use a fresh docker config for each step

### PR DESCRIPTION
**Pull request by yob**
This improves job isolation, preventing jobs from relying on authentication configured in other jobs.

This is based on PR #678, but we've removed the stack parameter because the v5.0.0 release is pending and we're comfortable making a small breaking change.

### CODE REVIEW

**Review by chloeruka — APPROVED** [review]
Straightforward change to improve step isolation. We've discussed this and decided it's better to make this (possibly) breaking change now as part of v5 release, and to bugfix it later if necessary. Original PR (#678) says this has been working well for them for quite some time.

**Comment by lox** [comment]
Couldn't we do this in the `environment` hook, or a bootstrap script?

**Comment by yob** [comment]
@lox are you suggesting this doesn't need to be baked in for everyone?

File: packer/linux/conf/buildkite-agent/hooks/pre-exit [hunk 1]
```diff
@@ -2,6 +2,11 @@
 
 set -eu -o pipefail
 
+# clean up our temporary docker config
+if [[ -n "${BUILDKITE_DOCKER_CONFIG_TEMP_DIRECTORY:-}" ]]; then
+  rmdir "$BUILDKITE_DOCKER_CONFIG_TEMP_DIRECTORY"
+fi
```
Comment (yob) [inline packer/linux/conf/buildkite-agent/hooks/pre-exit:8]: do you think it's worth checking if the directory exists as well? It shouldn't matter, but maybe there's a very small chance a customer build script will have deleted it fist (maybe by cleaning out the system tmp dir or something).

Comment (chloeruka) [reply packer/linux/conf/buildkite-agent/hooks/pre-exit:8]: Yeah, added a quick commit.

Comment (yob) [reply packer/linux/conf/buildkite-agent/hooks/pre-exit:8]: oh, this failed in the build too. Looks like it needs a `-rf`

Comment (yob) [reply packer/linux/conf/buildkite-agent/hooks/pre-exit:8]: interesting - it looks like `rmdir` will only delete empty directories. Maybe we need to use `rm`? Or I guess we could delete the contents first.

**Pull request merged**